ospfd/ospf_interface.c

```diff
@@ -1205,7 +1205,8 @@ u_char ospf_default_iftype(struct interface *ifp)
 {
 	if (if_is_pointopoint(ifp))
 		return OSPF_IFTYPE_POINTOPOINT;
-	else if (if_is_loopback(ifp))
+	else if (if_is_loopback(ifp) ||
+		 CHECK_FLAG(ifp->status, ZEBRA_INTERFACE_VRF_LOOPBACK))
```

Comment:
I would prefer, in lib/if.h:

#define if_is_vrf(ifp) CHECK_FLAG(ifp->status, ZEBRA_INTERFACE_VRF_LOOPBACK)

and then calling if_is_vrf(ifp) here.
ospf should not be peeking into interface specific details( that may or may not change in the future )